pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -271,20 +319,73 @@ public static String recoverSpecialCharacterInPropertyValue(String value) {
     return value.replace("\0\0", ",");
   }
 
-  private static PropertiesConfiguration createPropertiesConfiguration(boolean setIOFactory,
-      boolean setDefaultDelimiter) {
+  /**
+   * creates the instance of the {@link org.apache.commons.configuration2.PropertiesConfiguration}
+   * with custom IO factory based on kind {@link org.apache.commons.configuration2.PropertiesConfiguration.IOFactory}
+   * and legacy list delimiter {@link org.apache.commons.configuration2.convert.LegacyListDelimiterHandler}
+   *
+   * @param setDefaultDelimiter sets the default list delimiter.
+   * @param ioFactoryKind IOFactory kind
+   * @return PropertiesConfiguration
+   */
+  private static PropertiesConfiguration createPropertiesConfiguration(boolean setDefaultDelimiter,
+      PropertyIOFactoryKind ioFactoryKind) {
```

Comment:
allow to pass in null ioFactoryKind, so that we can keep previous logic intact, 
```
if (ioFactoryKind != null) {
  config.setIOFactory(createPropertyIOFactory(ioFactoryKind));
}
```